### ISSUE TITLE
Default index in the url

### ISSUE THREAD

**Issue by psaiz** [wtakase/kibana-own-home#1]
Hi Wataru, 

Very nice plugin!

I have a feature request. The main use case that I want to solve here is to have the  kibana index also on the url, so that I could shared it and people would be redirected to the same kibana index. At the moment, I solve this by setting different aliases on the machine, and then, based on the alias used, the proxy in front of kibana sets the X-Proxy-User variable that the kibana-own-home uses.

I've been playing with using the same alias, and then having different directories. I didn't manage to get this setup working because kibana needs to know the server.basePath, and in that case, it would be different. 

In other words, at the moment I have something like

```
<alias-1>/kibana
<alias-2>/kibana
...
<alias-n>/kibana
```
and I would like to move to something like
```
<host>/<prefix-1>/kibana
<host>/<prefix-2>/kibana
..
<host>/prefix-n>/kibana
```
Do you think that, the same way that the kibana-own-home modifies the elasticsearch.url, it could also modify the server.basepath?

Thanks, 
pablo

**Comment by wtakase** [comment]
> I have a feature request. The main use case that I want to solve here is to have the kibana index also on the url, so that I could shared it and people would be redirected to the same kibana index. 

I implemented the feature:
https://github.com/wtakase/kibana-own-home#set-default-kibanaindex-by-url

how about it?

About changing the server.basepath, I need some time to consider it.

**Comment by psaiz** [comment]
Hi Wataru, 

Thanks for your reply. The default index by url solves part of the issue. With it, you can put the entry point to the index that you want. Now, imagine the use case that I've been working on a particular view (or dashboard). If I take the URL directly, and send it to someone else, that will not work. They would not know in which kibana index I've been working. I'd have to build the link with the format of the set-default- kibanaindex-by-url by hand. 

Making the 'share' button modify the link according to the format would already be an improvement. 

Cheers,
 pablo

**Comment by wtakase** [comment]
Although it's still experimental, how about [this](https://github.com/wtakase/kibana-own-home/tree/explicit-kibana-index-url#use-explicit-kibanaindex-in-url-feature-experimental)?

**Comment by psaiz** [comment]
Hi Wataru,

This looks very promising! 

Do I understand correctly that apache then redirects to your plugin, instead of redirecting to kibana? 

In this case, is kibana still going through your plugin before going to elasticsearch?

I'll test this, with an apache configuration like.

'''
RewriteEngine on
RewriteRule  ^/?(.*)$ http://localhost:15601/$1 [L,P]
RequestHeader set X-Proxy-User %{REMOTE_USER}e
<Location /private1>
  AuthType shibboleth
  Require shib-attr ADFS_GROUP  egroup-private1
 </Location>
<Location /private2>
  AuthType shibboleth
  Require shib-attr ADFS_GROUP  egroup-private1
 </Location>

'''
So that people belonging to the egroup-private1 can see dashboard private1, and the group2 the egroup-private2

If this works, this is exactly what I had in mind! 

Thanks a lot,
pablo

**Comment by wtakase** [comment]
> Do I understand correctly that now your plugin listens on two ports: one to intercept the calls from apache to kibana, and the second from kibana to elasticsearch?

Yes.

> In kthe kibana configuration, did you leave the server.basepath as '/' ?

Yes.

I don't test enough, but the configuration such as follows:
httpd conf:
```
  RewriteEngine on
  RewriteRule  ^/?(.*)$ http://localhost:15601/$1 [L,P]

  RequestHeader set X-PROXY-USER %{REMOTE_USER}e

  <Location />
    AuthType shibboleth
    Require valid-user
  </Location>

  <Location /app/kibana>
    Order deny,allow
    Deny from all
  </Location>

  <Location /group1>
    AuthType shibboleth
    Require shib-attr ADFS_GROUP group1
  </Location>

  <Location /group2>
    AuthType shibboleth
    Require shib-attr ADFS_GROUP group2
  </Location>
```
and kibana.yml:
```
elasticsearch.url: http://localhost:19200
elasticsearch.requestHeadersWhitelist: [ x-proxy-user, cookie ]
own_home.elasticsearch.url: http://localhost:9200
own_home.session.secretkey: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
own_home.session.isSecure: false
own_home.local.enabled: false
own_home.ldap.enabled: true
own_home.ldap.url: ldap://localhost:389
own_home.ldap.userbase: ou=People,dc=localhost
own_home.ldap.rolebase: ou=Groups,dc=localhost
own_home.ldap.search_filter: '(|(cn=group1)(cn=group2))'
own_home.ldap.username_attribute: cn
own_home.ldap.rolename_attribute: cn
own_home.ldap.adfs: true
own_home.explicit_kibana_index_url.enabled: true
own_home.explicit_kibana_index_url.proxy.url: http://localhost:15601
```
Access => `frontendserver/group1` or `frontendserver/group2`

I hope this will work well.
